### PR TITLE
176 create a general textfield component

### PR DESCRIPTION
**Pull request by julianhahn**
Added a new textfield Vue component. A little showcase is added to the home.vue

### CODE REVIEW

File: packages/app/src/views/Home.vue [hunk 1]
```diff
@@ -1,27 +1,34 @@
 <template>
   <div class="m-auto">
     <pre>{{ user }}</pre>
+    <p>Suchefeld</p>
+    <Textfield class="mb-4" v-model="text" icon="search" placeholder="Titelsuche"></Textfield>
+    <p>Textfield</p>
+    <Textfield class="mb-4" v-model="text" label="Title" placeholder="Titelsuche"></Textfield>
```
Comment (tcoro) [inline packages/app/src/views/Home.vue:7]:
```suggestion
    <Textfield class="mb-4" v-model="text" label="Title" placeholder="Titelsuche"/>
```

File: packages/app/src/views/Home.vue [hunk 2]
```diff
@@ -1,27 +1,34 @@
 <template>
   <div class="m-auto">
     <pre>{{ user }}</pre>
+    <p>Suchefeld</p>
+    <Textfield class="mb-4" v-model="text" icon="search" placeholder="Titelsuche"></Textfield>
```
Comment (tcoro) [inline packages/app/src/views/Home.vue:5]:
```suggestion
    <Textfield class="mb-4" v-model="text" icon="search" placeholder="Titelsuche"/>
```

File: packages/app/src/components/uiBlocks/Textfield.vue [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+<script lang="ts">
+import { defineComponent } from 'vue';
+
+export default defineComponent({
+  props: {
+    placeholder: {
+      type: String,
+      required: true,
+    },
+    modelValue: {
```
Comment (tcoro) [inline packages/app/src/components/uiBlocks/Textfield.vue:28]: what does this do?
I would suggest changing the name, just from plain reading the sourcecode I couldn't understand the function of this.
Is this the value entered by a user into the textfield?
How about

```suggestion
    inputvalue: {
```
or

```suggestion
    input: {
```

File: packages/app/package.json [hunk 4]
```diff
@@ -11,9 +11,10 @@
   },
   "dependencies": {
     "@feathersjs/adapter-commons": "4.5.11",
+    "@feathersjs/authentication-client": "4.5.11",
     "@feathersjs/feathers": "4.5.11",
     "@feathersjs/socketio-client": "4.5.11",
-    "@feathersjs/authentication-client": "4.5.11",
+    "@fortawesome/fontawesome-free": "^5.15.1",
```
Comment (tcoro) [inline packages/app/package.json:17]: we should communicate that we use another font in the next meeting, so this font is used unified, else we will have different fonts.
Or maybe enforce this font for everyone in some way

**Review by tcoro — CHANGES_REQUESTED** [review]
Looks great, just some minor changes/ideas

File: packages/app/src/views/Home.vue (hunk 1 above)
Comment (julianhahn) [reply packages/app/src/views/Home.vue:7]: fixed in 43ad87abf73eacdd5789a326654f80382c57979e

File: packages/app/src/views/Home.vue (hunk 2 above)
Comment (julianhahn) [reply packages/app/src/views/Home.vue]: fixed in 43ad87abf73eacdd5789a326654f80382c57979e

File: packages/app/src/views/Home.vue (hunk 1 above)
Comment (julianhahn) [reply packages/app/src/views/Home.vue]: fixed in 43ad87abf73eacdd5789a326654f80382c57979e

File: packages/app/package.json (hunk 4 above)
Comment (julianhahn) [reply packages/app/package.json:17]: we do not change the font in this branch. We use the fortawesome plugin to show icons in the textfield.

File: packages/app/src/components/uiBlocks/Textfield.vue (hunk 3 above)
Comment (julianhahn) [reply packages/app/src/components/uiBlocks/Textfield.vue:30]: The v-model command simly replace the command 'v-model' with the text v-bind:value="searchText"
v-on:input="searchText = $event.target.value". It gives a prop an wants to listen on it's change.
further reading on:
https://v3.vuejs.org/guide/component-custom-events.html#handling-v-model-modifiers

**Review by schnickalodeon — APPROVED** [review]
> Looks nice 👍

File: packages/app/package.json (hunk 4 above)
Comment (tcoro) [reply packages/app/package.json:17]: oh okay

**Review by tcoro — APPROVED** [review]
looks good

File: packages/app/src/components/uiBlocks/Textfield.vue (hunk 3 above)
Comment (tcoro) [reply packages/app/src/components/uiBlocks/Textfield.vue:30]: Do you think a small comment might be good here?

Comment (julianhahn) [reply packages/app/src/components/uiBlocks/Textfield.vue:30]: Not quite sure. This is the standard to read values from child components with vue 3 (to my knowlege). So my comment would just say, that it has to be like this.. no quite usefull. A link to the source as a comment might be outdated quickly.

**Pull request merged**